bodo/pandas/base.py

```diff
@@ -239,6 +260,101 @@ def read_iceberg(
+    usecols: UsecolsArgType = None,
+    # General Parsing Configuration
+    dtype: DtypeArg | None = None,
+    engine: CSVEngine | None = None,
+    converters: Mapping[Hashable, pt.Callable] | None = None,
+    true_values: list | None = None,
+    false_values: list | None = None,
+    skipinitialspace: bool = False,
+    skiprows: list[int] | int | pt.Callable[[Hashable], bool] | None = None,
+    skipfooter: int = 0,
+    nrows: int | None = None,
+    # NA and Missing Data Handling
+    na_values: Hashable
+    | Iterable[Hashable]
+    | Mapping[Hashable, Iterable[Hashable]]
+    | None = None,
+    keep_default_na: bool = True,
+    na_filter: bool = True,
+    verbose: bool | lib.NoDefault = lib.no_default,
+    skip_blank_lines: bool = True,
+    # Datetime Handling
+    parse_dates: bool | Sequence[Hashable] | None = None,
+    infer_datetime_format: bool | lib.NoDefault = lib.no_default,
+    keep_date_col: bool | lib.NoDefault = lib.no_default,
+    date_parser: pt.Callable | lib.NoDefault = lib.no_default,
+    date_format: str | dict[Hashable, str] | None = None,
+    dayfirst: bool = False,
+    cache_dates: bool = True,
+    # Iteration
+    iterator: bool = False,
+    chunksize: int | None = None,
+    # Quoting, Compression, and File Format
+    compression: CompressionOptions = "infer",
+    thousands: str | None = None,
+    decimal: str = ".",
+    lineterminator: str | None = None,
+    quotechar: str = '"',
+    quoting: int = csv.QUOTE_MINIMAL,
+    doublequote: bool = True,
+    escapechar: str | None = None,
+    comment: str | None = None,
+    encoding: str | None = None,
+    encoding_errors: str | None = "strict",
+    dialect: str | csv.Dialect | None = None,
+    # Error Handling
+    on_bad_lines: str = "error",
+    # Internal
+    delim_whitespace: bool | lib.NoDefault = lib.no_default,
+    low_memory: bool = _c_parser_defaults["low_memory"],
+    memory_map: bool = False,
+    float_precision: pt.Literal["high", "legacy"] | None = None,
+    storage_options: StorageOptions | None = None,
+    dtype_backend: DtypeBackend | lib.NoDefault = lib.no_default,
+) -> BodoDataFrame:
+    func = "def bodo_read_csv(filepath"
+    if names != lib.no_default:
+        func += ", names"
+    if usecols != None:
+        func += ", usecols"
+    if parse_dates != None:
+        func += ", parse_dates"
+    func += "):\n"
+    func += "    return pd.read_csv(filepath"
+    func_args = []
+    if names != lib.no_default:
+        func += ", names=names"
+        func_args.append(names)
+    if usecols != None:
+        func += ", usecols=usecols"
+        func_args.append(usecols)
+    if parse_dates != None:
+        func += ", parse_dates=parse_dates"
+        func_args.append(parse_dates)
+    func += ")\n"
+    csv_func = bodo_spawn_exec(func, {"pd": pd}, {}, __name__)
+    jit_csv_func = bodo.jit(csv_func)
+    return jit_csv_func(filepath_or_buffer, *func_args)
```

Comment:
Is it possible to just pass the parameters to a JIT function and avoid execing a function generated from string? `no_default` would need handling though.